addons/html_builder/static/src/core/anchor/anchor_plugin.js

```diff
@@ -0,0 +1,132 @@
+import { escape } from "@web/core/utils/strings";
+
+const anchorSelector = ":not(p).oe_structure > *, :not(p)[data-oe-type=html] > *";
+const anchorExclude =
+    ".modal *, .oe_structure .oe_structure *, [data-oe-type=html] .oe_structure *, .s_popup";
+
+export function canHaveAnchor(element) {
+    return element.matches(anchorSelector) && !element.matches(anchorExclude);
+}
+
+export class AnchorPlugin extends Plugin {
+    static id = "anchor";
+    static dependencies = ["history"];
+    static shared = ["createOrEditAnchorLink"];
+    resources = {
+        on_cloned_handlers: this.onCloned.bind(this),
+        get_options_container_top_buttons: withSequence(
+            0,
+            this.getOptionsContainerTopButtons.bind(this)
+        ),
+    };
+
+    onCloned({ cloneEl }) {
+        const anchorEls = getElementsWithOption(cloneEl, anchorSelector, anchorExclude);
+        anchorEls.forEach((anchorEl) => this.deleteAnchor(anchorEl));
+    }
+
+    getOptionsContainerTopButtons(el) {
+        if (!canHaveAnchor(el)) {
+            return [];
+        }
+
+        return [
+            {
+                class: "fa fa-fw fa-link oe_snippet_anchor btn btn-outline-info",
+                title: _t("Create and copy a link targeting this block or edit it"),
+                handler: this.createOrEditAnchorLink.bind(this),
+            },
+        ];
+    }
+
+    // TODO check if no other way when doing popup options.
+    isModal(element) {
+        return element.classList.contains("modal");
+    }
+
+    setAnchorName(element, value) {
+        if (value) {
+            element.id = value;
+            if (!this.isModal(element)) {
+                element.dataset.anchor = true;
+            }
+        } else {
+            this.deleteAnchor(element);
+        }
+        this.dependencies.history.addStep();
+    }
+
+    createAnchor(element) {
+        const titleEls = element.querySelectorAll("h1, h2, h3, h4, h5, h6");
+        const title = titleEls.length > 0 ? titleEls[0].innerText : element.dataset.name;
+        const anchorName = this.formatAnchor(title);
+
+        let n = "";
+        while (this.document.getElementById(anchorName + n)) {
+            n = (n || 1) + 1;
+        }
+
+        this.setAnchorName(element, anchorName + n);
+    }
+
+    deleteAnchor(element) {
+        element.removeAttribute("data-anchor");
+        element.removeAttribute("id");
+    }
+
+    getAnchorLink(element) {
+        const pathName = this.isModal(element) ? "" : this.document.location.pathname;
+        return `${pathName}#${element.id}`;
+    }
+
+    async createOrEditAnchorLink(element) {
+        if (!element.id) {
+            this.createAnchor(element);
+        }
+        const anchorLink = this.getAnchorLink(element);
+        await browser.navigator.clipboard.writeText(anchorLink);
+        const message = markup(_t("Anchor copied to clipboard<br>Link: %s", escape(anchorLink)));
```

Comment:
We should avoid introducing more cases of the bad pattern, the safe version is not really more complex, more flexible for translators, and safer:
```suggestion
    const message = _t("Anchor copied to clipboard%(br)sLink: %(anchorLink)s", { anchorLink, br: markup`<br>` });
```
(I didn't re-test the syntax but this should be working unless I made a typo)